src/dune_engine/dune_project.ml

```diff
@@ -791,6 +800,7 @@ let parse ~dir ~lang ~opam_packages ~file =
      let generate_opam_files =
        Option.value ~default:false generate_opam_files
      in
+     let always_add_cflags = Option.value ~default:true always_add_cflags in
```

Comment:
I would suggest to pick the default based on the dune version. It should be false for 3.0 and above.